cub/test/catch2_test_device_reduce_env.cu

```diff
@@ -334,7 +336,9 @@ C2H_TEST("Device sum uses environment", "[reduce][device]", requirements)
 
   // To check if a given algorithm implementation is used, we check if associated kernels are invoked.
   auto kernels = [&]() {
-    if constexpr (std::is_same_v<determinism_t, cuda::execution::determinism::run_to_run_t>)
+    // TODO(gevtushenko): split `not_guaranteed` kernels once atomic reduce is merged
+    if constexpr (std::is_same_v<determinism_t, cuda::execution::determinism::run_to_run_t>
+                  || std::is_same_v<determinism_t, cuda::execution::determinism::not_guaranteed_t>)
```

Comment:
Important: the `not_guaranteed_t` path is handled in the next branch with `is_non_deterministic_v`, so this change should be undone. Also, the TODO comment should be deleted.